controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala

```diff
@@ -28,24 +31,59 @@ class ProjectFileController(wdlService: ProjectFileService)(
         }
       },
       path("files") {
-        post {
-          entity(as[ProjectUpdateFileRequest]) {
-            request =>
-              onComplete(for {
-                validateResponse <- wdlService.validateFile(FileContent(request.projectFile.content))
-                uploadResponse <- wdlService.uploadFile(request.project, request.projectFile, request.version)
-              } yield {
-                (validateResponse, uploadResponse) match {
-                  case (Right(_), Right(response)) => StatusCodes.OK.intValue -> response
-                  case (Left(_), Right(response))  => StatusCodes.Created.intValue -> response
-                  case (_, Left(response))         => StatusCodes.UnprocessableEntity.intValue -> response.message
+        concat(
+          post {
+            entity(as[ProjectUpdateFileRequest]) {
+              request =>
+                onComplete(for {
+                  validateResponse <- wdlService.validateFile(FileContent(request.projectFile.content))
+                  uploadResponse <- wdlService.uploadFile(request.project, request.projectFile, request.version)
+                } yield {
+                  (validateResponse, uploadResponse) match {
+                    case (Right(_), Right(response)) => StatusCodes.OK.intValue -> response
+                    case (Left(_), Right(response))  => StatusCodes.Created.intValue -> response
+                    case (_, Left(response))         => StatusCodes.UnprocessableEntity.intValue -> response.message
+                  }
+                }) {
+                  case Success((status, message)) => complete(status, message)
+                  case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
+                }
+            }
+          },
+          get {
+            parameter('projectId.as[String], 'path.as[String], 'version.as[String]) {
```

Comment:
Didn't we agree that `version` query parameter should be optional? 
Also, this way of declaring parameters is deprecated in Akka-http now as far as I remember